### PR TITLE
[LOOP-4793] Beginning XCUI Tests

### PR DESCRIPTION
**Pull request by Camji55**
https://tidepool.atlassian.net/browse/LOOP-4793

### CODE REVIEW

File: LoopUITests/Screens/HomeScreen.swift [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        waitForExistence(deliverBolusButton)
+        deliverBolusButton.forceTap()
+    }
+    
+    func verifyOcclusionAlert() {
+//        waitForExistence(notification)
+//        notification.tap()
+//        waitForExistence(bolusIssueNotificationTitle)
+//        app.activate()
+        #warning("FIXME")
```
Comment (nhamming) [inline LoopUITests/Screens/HomeScreen.swift]: should this stay?

File: LoopUITests/Screens/HomeScreen.swift [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    }
+    
+    func enterPasscode() {
+        waitForExistence(passcodeEntry)
+        passcodeEntry.tap()
+        springboardApp.typeText("1\n")
+//        sleep(1)
+//        waitForExistence(springboardKeyboardDoneButton)
+//        springboardKeyboardDoneButton.tap()
+    }
```
Comment (nhamming) [inline LoopUITests/Screens/HomeScreen.swift]: should this stay?

**Review by nhamming — APPROVED** [review]
LGTM

**Comment by nhamming** [comment]
@Camji55 I made the assumption that these UI tests run for both Tidepool and DIY without issue. Safe assumption?

File: LoopUITests/LoopUITestPlan.xctestplan [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      "name" : "Configuration 1",
+      "options" : {
+        "environmentVariableEntries" : [
+          {
+            "key" : "appName",
+            "value" : "Loop"
+          },
+          {
+            "key" : "bundleIdentifier",
+            "value" : "org.tidepool.Loop"
```
Comment (ps2) [inline LoopUITests/LoopUITestPlan.xctestplan]: I don't think this is right. Do we need some way of changing this for the different builds?

File: LoopUITests/Screens/BaseScreen.swift [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    init(app: XCUIApplication) {
+        self.app = app
+        self.springboardApp = XCUIApplication(bundleIdentifier:"com.apple.springboard")
+        self.bundleIdentifier = Bundle.main.bundleIdentifier
+    }
+    
+    func deleteApp() {
+        XCUIApplication().terminate()
+
+        let icon = springboardApp.icons["Tidepool Loop"]
```
Comment (ps2) [inline LoopUITests/Screens/BaseScreen.swift]: This is testing the Tidepool Loop version of Loop, but this is in shared code.  We may need to find a way to make this not Tidepool Specific

File: LoopUITests/Screens/SystemSettingsScreen.swift [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+//
+
+import XCTest
+
+final class SystemSettingsScreen: BaseScreen {
+    
+    // MARK: Elements
+    
+    var loopCell: XCUIElement {
+        app.cells["Tidepool Loop"]
```
Comment (ps2) [inline LoopUITests/Screens/SystemSettingsScreen.swift]: Another Tidepool specific string in shared code.

**Review by ps2 — CHANGES_REQUESTED** [review]
I think we need to find a way to genericize the bundle id and some strings, as this is a shared code area.

File: LoopUITests/Screens/BaseScreen.swift (hunk 4 above)
Comment (Camji55) [reply LoopUITests/Screens/BaseScreen.swift]: We can derive this. Will update.

File: LoopUITests/Screens/SystemSettingsScreen.swift (hunk 5 above)
Comment (Camji55) [reply LoopUITests/Screens/SystemSettingsScreen.swift]: We can derive this. Will update.

File: LoopUITests/Screens/HomeScreen.swift (hunk 1 above)
Comment (Camji55) [reply LoopUITests/Screens/HomeScreen.swift]: Fixed

File: LoopUITests/Screens/HomeScreen.swift (hunk 2 above)
Comment (Camji55) [reply LoopUITests/Screens/HomeScreen.swift]: Fixed

File: LoopUITests/LoopUITestPlan.xctestplan (hunk 3 above)
Comment (Camji55) [reply LoopUITests/LoopUITestPlan.xctestplan]: Fixed

File: Loop.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -5609,6 +5608,14 @@
 /* End XCConfigurationList section */
 
 /* Begin XCRemoteSwiftPackageReference section */
+		845C74332B7D686000F71F90 /* XCRemoteSwiftPackageReference "LoopUITestingKit" */ = {
+			isa = XCRemoteSwiftPackageReference;
+			repositoryURL = "https://github.com/tidepool-org/LoopUITestingKit.git";
```
Comment (ps2) [inline Loop.xcodeproj/project.pbxproj]: Is this needed? It looks like it introduces a dependency on a Tidepool repo to Loop.

File: LoopTests/DIYLoopUnitTestPlan.xctestplan [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    {
+      "target" : {
+        "containerPath" : "container:NightscoutService\/NightscoutService.xcodeproj",
+        "identifier" : "A91BAC2322BC691A00ABF1BB",
+        "name" : "NightscoutServiceKitTests"
+      }
+    },
+    {
+      "target" : {
+        "containerPath" : "container:..\/Common\/TidepoolService\/TidepoolService.xcodeproj",
```
Comment (ps2) [inline LoopTests/DIYLoopUnitTestPlan.xctestplan]: More Tidepool references?

**Review by ps2 — COMMENTED** [review]
Looks like there are still some Tidepool references?

File: Loop.xcodeproj/project.pbxproj (hunk 6 above)
Comment (Camji55) [reply Loop.xcodeproj/project.pbxproj]: Nope, not needed!

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**